src/NBitcoin/Block.cs

```diff
@@ -220,6 +220,9 @@ public partial class Block : IBitcoinSerializable
 
         private BlockHeader header;
 
+        /// <summary>The size of the block in bytes, the block must be serialized for this property to be set.</summary>
+        public long? BlockSize { get; protected set; }
```

Comment:
The `BlockSize` is unqualified and public - i.e. it is visible to any code to make assumptions as to the `TransactionOptions` and `Network` the size is for.

Reply:
I didn't understand the comment.

Reply:
E.g. Recording the BlockSizeNetwork and BlockSizeTransactionOptions would allow to check what the recorded BlockSize contains - i.e. qualify it. Not that I am suggesting that would be the best or safest approach.